xml/map.go

```diff
@@ -0,0 +1,50 @@
+package xml
+
+import (
+	"bytes"
+)
+
+// mapKey is the default member wrapper tag name for XML Map type
+const mapKey = "entry"
+
+// Map represents the encoding of a XML map type
+type Map struct {
+	w          *bytes.Buffer
+	scratch    *[]byte
+	openTagFn  func()
+	closeTagFn func()
+}
+
+// newMap returns a map encoder which sets the default map
+// entry wrapper to `entry`.
+//
+// for eg. someMap : {{key:"abc", value:"123"}} is represented as
+// <someMap><entry><key>abc<key><value>123</value></entry><member>value2</member></someMap>
+func newMap(w *bytes.Buffer, scratch *[]byte) *Map {
+	var openTagFn = func() {
```

Comment:
i think this example is wrong.  It probably should strip off the `<someMap>` tags

Reply:
`someMap` is the name of the member here. This example in accordance to https://awslabs.github.io/smithy/1.0/spec/core/xml-traits.html#wrapped-map-serialization